agent/config/runtime.go

```diff
@@ -1566,6 +1570,24 @@ type RuntimeConfig struct {
 	EnterpriseRuntimeConfig
 }
 
+type AutoConfig struct {
+	Enabled         bool
+	IntroToken      string
+	IntroTokenFile  string
+	ServerAddresses []string
+	DNSSANs         []string
+	IPSANs          []net.IP
+	Authorizer      AutoConfigAuthorizer
+}
```

Comment:
Not directly related to your PR, but I have been wondering if we would be able to start moving more of the "Runtime" structs into the packages that will be receiving those values. That way we should be able to drop one of the 3 permutations of config structs.

Reply:
I was thinking the same thing but didn't have time to do it here. I think there may be issues with circular dependencies but nothing insurmountable. 

We copy config values around so many places unnecessarily.